platform/email/platform-email-impl/src/main/java/org/codice/ddf/platform/email/impl/SmtpClientImpl.java

```diff
@@ -0,0 +1,132 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.platform.email.impl;
+
+import static org.apache.commons.lang.Validate.isTrue;
+import static org.apache.commons.lang.Validate.notEmpty;
+import static org.apache.commons.lang.Validate.notNull;
+
+import java.util.Properties;
+
+import javax.mail.Authenticator;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.PasswordAuthentication;
+import javax.mail.Session;
+import javax.mail.Transport;
+
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.platform.email.SmtpClient;
+
+/**
+ * Supports unauthenticated connections to a mail server and username/password authenticated connections
+ * over TLS. If the username is not blank and the password is not empty, then auth/tls will be used.
+ *
+ * The methods {@link #createSession()} and {@link #send(Message)} are thread-safe.
+ *
+ * <p>
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ * </p>
+ */
+public class SmtpClientImpl implements SmtpClient {
+
+    private static final String SMTP_HOST_PROPERTY = "mail.smtp.host";
+
+    private static final String SMTP_PORT_PROPERTY = "mail.smtp.port";
+
+    private static final String SMTP_AUTH_PROPERTY = "mail.smtp.auth";
+
+    private static final String SMTP_START_TLS_ENABLE_PROPERTY = "mail.smtp.starttls.enable";
+
+    private static final String TRUE = "true";
+
+    private static final String FALSE = "false";
+
+    private String hostName;
+
+    private Integer portNumber;
+
+    private String userName;
+
+    private String password;
+
+    /**
+     * Set the username for the email server.
+     *
+     * @param userName may be null or blank
+     */
+    public synchronized void setUserName(String userName) {
```

Comment:
So was the synchronization on the class a way to throttle requests? If so, I'm not sure this is the right way to achieve that.